### ISSUE TITLE
Question about fine-tuning scGPT_Human model when a large proportion of genes are missing from the vocabulary

### ISSUE THREAD

**Issue by wolfQK** [RCHENLAB/scGPT_fineTune_protocol#1]
Hello,

I am trying to fine-tune the scGPT_Human model using my own dataset, but I found that a large proportion of genes in my dataset are not present in the model’s vocabulary.

## Background: 
My dataset was generated using the following GENCODE annotation file:

> ##description: evidence-based annotation of the human genome (GRCh38), version 44 (Ensembl 110)  
> ##provider: GENCODE  
> ##: gencode-help@ebi.ac.uk  
> ##format: gtf  
> ##date: 2023-03-01  

When matching my dataset’s gene names with the vocabulary of scGPT_Human, I got:
> match 24498 / 38606 genes in vocabulary of size 60697

That means **36.54% of my genes are not in the vocabulary** !

Most of the missing genes are unannotated lncRNAs or novel transcripts whose gene_name equals their Ensembl ID (ENSG...), for example:
> ENSG00000290826
> ENSG00000238009
> ENSG00000239945
> ENSG00000239906
> ENSG00000241860
> ENSG00000241599
> ENSG00000286448

## Question
Given this situation:
- **Can I still fine-tune the model as usual**, even though 36.54% of the genes are missing from its vocabulary?  
- Or would it be better to **train a new model from scratch** with a vocabulary based on my own dataset, so that all genes are included?  

@Ding3LI  @RuiLuo111  @RCHENLAB 

I would greatly appreciate your advice on the best approach to proceed.

Thanks!

**Comment by Ding3LI** [comment]
Hi @wolfQK , I hope your research is going well. To the best of my understanding, even with 36.54% of genes missing, you should still be able to train your classifier effectively.

In response to your questions:

* Please take advantage of fine-tuning using the pre-trained whole-body model rather than training from scratch. This is a large model that takes an extremely long time to converge and requires a large amount of data due to the number of parameters involved.
* I believe that ~60% gene coverage will still allow the model to learn well for your custom dataset. In our curated retina dataset, the gene coverage is approximately 75%. However, there is a crucial parameter, "**max_seq_len**", which controls the effective input length of the gene matrix. As a result, even with 36.54% of genes missing from the current gene library, the model may not be using all detected genes in practice.

Overall, **_YES_**, please proceed with fine-tuning your custom model. And, I do NOT recommend training from scratch.

Please feel free to share any comments and let me know if you have further questions. Thanks very much! Happy New Year!

**Comment by wolfQK** [comment]
@Ding3LI 

Thank you very much for your detailed explanation and helpful suggestions.
I truly appreciate your time and guidance. Thanks again, and Happy New Year to you too !
